onnxruntime_extensions/tools/add_HuggingFace_CLIPImageProcessor_to_model.py

```diff
@@ -0,0 +1,171 @@
+        layout = 'CHW'
+        steps.append(Normalize(mean_std, layout=layout))
+
+    steps.append(Unsqueeze([0]))  # add batch dim
+
+    return steps
+
+
+def clip_image_processor(model_file: Path, output_file: Path, **kwargs):
+    """
+    Used for models like stable-diffusion. should be compatible with 
+    https://github.com/huggingface/transformers/blob/main/src/transformers/models/clip/image_processing_clip.py
+
+    It's similar to 'CLIP' image processor, and aligns with the HuggingFace class name.
+
+    A typical usage example is used in Stable diffusion.
+
+    :param model_file: The input model file path.
+    :param output_file: The output file path, where the finalized model saved to.
+    :param Kwargs
+        onnx_opset: The opset version of onnx model, default(18).
+        do_convert_rgb: Convert image from BGR to RGB. default(True)
+        do_resize: Resize the image's (height, width) dimensions to the specified `size`. default(True)
+        size: The shortest edge of the image is resized to size. Default(224)
+        resample: An optional resampling filter. Default(cubic)
+        do_center_crop: Whether to center crop the image to the specified `crop_size`. Default(True)
+        crop_size: Size of the output image after applying `center_crop`. Default(224)
+        do_rescale: Whether to rescale the image by the specified scale (rescale_factor). Default(True)
+        rescale_factor: Scale factor to use if rescaling the image. Default(1/255)
+        do_normalize: Whether to normalize the image. Default(True)
+        image_mean: Mean values for image normalization. Default([0.485, 0.456, 0.406])
+        image_std: Standard deviation values for image normalization. Default([0.229, 0.224, 0.225])
+
+
+    """
+    args = Dict2Class(kwargs)
+    # Load model
+    model = onnx.load(str(model_file.resolve(strict=True)))
+    inputs = [create_named_value("image", onnx.TensorProto.UINT8, ["num_bytes"])]
+
+    pipeline = PrePostProcessor(inputs, args.opset)
+
+    preprocessing = image_processor(args)
+
+    pipeline.add_pre_processing(preprocessing)
+
+    new_model = pipeline.run(model)
+    onnx.save_model(new_model, str(output_file.resolve()))
+    print(f"Updated model saved to {output_file}")
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        os.path.basename(__file__),
+        description="""Add CLIPImageProcessor to a model.
+
+        The updated model will be written in the same location as the original model, 
+        with '.onnx' updated to '.with_clip_processor.onnx'
+
+        Example usage:
+            object detection:
+            - python -m onnxruntime_extensions.tools.add_HuggingFace_CLIPImageProcessor_to_model model.onnx  
+        """,
+    )
+
+    parser.add_argument(
+        "--opset", type=int, required=False, default=18,
+        help="ONNX opset to use. Minimum allowed is 18. Opset 18 is required for Resize with anti-aliasing.",
```

Comment:
The minimum allowed is still 16 isn't it as the pre/post processing Step implementations used that as the base opset? We want people to use 18+ for the anti-aliasing but updating the model should still work with opset 16.

Reply:
Yes. Minimum allowed is 16 but we can set the default opset as 18.